lib/model.js

```diff
@@ -1494,8 +1494,14 @@ function _decorateDiscriminatorIndexOptions(model, indexOptions) {
   if (model.baseModelName != null && indexOptions.unique &&
       !('partialFilterExpression' in indexOptions) &&
       !('sparse' in indexOptions)) {
+    
+    const value = (
+      model.schema.discriminatorMapping &&
```

Comment:
I'm not really familiar with `mongoose` internals, maybe this `&&` check isn't necessary.